### DISCUSSION TITLE
Change filename of .tex according to profile when keeptex: true is used

### DISCUSSION THREAD

**Discussion by skwde** [quarto-dev/quarto-cli#10224] [Q&A] [answered]
### Description

This is probably related to <https://github.com/quarto-dev/quarto-cli/issues/1294> and kind of a follow up to the old discussion <https://github.com/quarto-dev/quarto-cli/discussions/2802>.

When using different profiles to generate e.g. exercises + solutions together with their tex source the name specified in the profile via `output-file` is only applied to the pdf even when we explicitly set `keeptex: true`.

I use quarto version `1.4.553` on debian bookworm in a container.

**Comment by mcanouil** [comment, marked as answer]
The file you get with `keep-tex` is not supposed to be used for anything but debugging.
It's an intermediate document.

If you want the final LaTeX file then ask for it.

```yaml
format:
  pdf: default
  latex: default
```

**Reply by skwde** [reply]
nice, exactly what I was looking for. Thanks!

Though I cannot find documentation on it. 
Certainly it seems not to be mentioned under Formats in the reference. (https://quarto.org/docs/reference/)
Would you mind adding a link here?

**Reply by mcanouil** [reply]
See <https://pandoc.org>.

**Reply by cderv** [reply]
> The file you get with keep-tex is not supposed to be used for anything but debugging.
It's an intermediate document.

Not 100% the case as I believe with Quarto Journals we use `keep-tex: true` the tex file and its resources as usually you want to send that to publishers. Though they are pdf format because you also want to see the PDF file produced. 

Setting both `latex` and `pdf` formats means two quarto rendering, while the `.tex` will be the same. 

We also document `keep-tex` as a way to produce a `.tex` file in addition to the pdf preview

@skwde this is documented here BTW https://quarto.org/docs/output-formats/pdf-basics.html#latex-output 
Going though Pandoc documentation is a bit hard as we try to document most of what is relevant in quarto.org, or link Pandoc doc when needed. Though, this is a good read if you want to understannd how all this works !  

Anyhow, even if most of the times the intermediate can be used for debugging, it could make sense that `output-file` also applies to the intermediate when `keep-*` is set to ture. 

This is currently an open enhancement not yet planned
- https://github.com/quarto-dev/quarto-cli/issues/6045

**Reply by mcanouil** [reply]
I missed that request.

Maybe all `keep-` should respect `output-file` then.

**Reply by skwde** [reply]
Thanks for the explanation and links.

Sorry for missing the related issue. I just gave my thumbs up there :). 

I looked over the pdf basics a while back and didn't remember the latex format there.

**Reply by cderv** [reply]
> Maybe all keep- should respect output-file then.

Yes that is the idea. 

> I looked over the pdf basics a while back and didn't remember the latex format there.

Could be new, we update our doc regularly

**Reply by skwde** [reply]
Could as well be that I just forgot, due to the many nice features of quarto.

Getting starting with quarto can at times be a bit overwhelming ;) (in a positive way)

**Reply by mcanouil** [reply]
I don't think `latex` as a format is explicitly mentioned in Quarto.

Quarto uses Pandoc as the last piece in the workflow (or nearly) so you can use Pandoc's formats.
Of course Quarto cannot guarantee all features works for all formats supported by Pandoc, but in this particular case, "latex" is basically a shorter version of the "pdf" format workflow, so it is unlikely to cause issues.
